Stripe/PublicHeaders/STPPaymentIntent.h

```diff
@@ -0,0 +1,98 @@
+//
+//  STPPaymentIntent.h
+//  Stripe
+//
+//  Created by Daniel Jackson on 6/27/18.
+//  Copyright © 2018 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "STPAPIResponseDecodable.h"
+#import "STPPaymentIntentEnums.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ A PaymentIntent tracks the process of collecting a payment from your customer.
+
+ @see https://stripe.com/docs/api#payment_intents
+ @see https://stripe.com/docs/payments/dynamic-authentication
+ */
+@interface STPPaymentIntent : NSObject<STPAPIResponseDecodable>
+
+/**
+ You cannot directly instantiate an `STPPaymentIntent`. You should only use one that
+ has been returned from an `STPAPIClient` callback.
+ */
+- (instancetype)init __attribute__((unavailable("You cannot directly instantiate an STPPaymentIntent. You should only use one that has been returned from an STPAPIClient callback.")));
+
+/**
+ The Stripe ID of the PaymentIntent.
+ */
+@property (nonatomic, readonly) NSString *stripeId;
+
+/**
+ The client secret used to fetch this PaymentIntent
+ */
+@property (nonatomic, readonly) NSString *clientSecret;
+
+/**
+ The amount associated with the PaymentIntent
+ */
+@property (nonatomic, readonly) NSNumber *amount;
+
+/**
+ If status is canceled, when the PaymentIntent was canceled.
+ */
+@property (nonatomic, nullable, readonly) NSDate *canceledAt;
+
+/**
+ Capture method of this PaymentIntent
+ */
+@property (nonatomic, readonly) STPPaymentIntentCaptureMethod captureMethod;
+
+/**
+ Confirmation method of this PaymentIntent
+ */
+@property (nonatomic, readonly) STPPaymentIntentConfirmationMethod confirmationMethod;
+
+/**
+ When the PaymentIntent was created.
+ */
+@property (nonatomic, nullable, readonly) NSDate *created;
+
+/**
+ The currency associated with the PaymentIntent.
```

Comment:
Our docs nicely mention it's the ISO currency code, I wonder if we mention that elsewhere we reference `currency`:

"Three-letter ISO currency code, in lowercase. Must be a supported currency."

https://www.iso.org/iso-4217-currency-codes.html
https://stripe.com/docs/currencies